### PR TITLE
feat: kanban + list-row + rail-drop-slot + lane-timeline (Dispatch primitives)

### PR DESCRIPTION
**Pull request by alioftech**
## Summary

Adds four big composed components to `@medalsocial/meda` — built for the upcoming **Dispatch** dashboard but designed generically for reuse across `pilot-talk`, `auto-cs`, and `apps/web`. Plus two supporting `components/ui/` primitives needed as transitive deps.

## New exports

| Subpath | Component | Notes |
|---|---|---|
| `./kanban` | `KanbanBoard` | Lifted from `apps/web/src/components/kanban-board/` (466 lines). Decoupled from `apps/web` internals — see below. |
| `./list` | `ListRow`, `ListCell` | Lifted from `apps/web/src/components/list-view/list-row.tsx`. |
| `./shell` | `RailDropSlot` | New generic `@dnd-kit/core` drop target. Three states: idle / hover / rejected / disabled. Pairs with `KanbanBoard` for drag-to-assign UX. |
| `./timeline` | `LaneTimeline` | New swimlane Gantt. Owns time-window math, range selector (1h/6h/24h/7d), date switcher, tick row, now-line, legend internally. Consumer passes pre-grouped \`Lane[]\`. |

## New primitives in \`components/ui/\` (base-ui-backed, mirroring \`apps/web\` exactly)

- `Checkbox` — wraps `@base-ui/react/checkbox`. Indeterminate support, `data-slot` markers, full dark-mode + WCAG 2.5.5 touch-target classes, character-for-character match to `apps/web/src/components/ui/checkbox.tsx`.
- `Collapsible` (+ `CollapsibleTrigger` + `CollapsibleContent`) — pure pass-through wrappers over `@base-ui/react/collapsible` with `data-slot` markers.

## Decoupling done during the kanban extraction

\`KanbanBoard\` and \`KanbanColumn\` originally depended on three \`apps/web\` internals. Extraction required generalizing each:

1. **Paraglide i18n → controlled \`labels\` prop.** Removed \`import { m } from '@/paraglide/messages';\`. \`KanbanBoardProps\` now accepts \`labels?: Partial<KanbanLabels>\`; default English values live in \`defaultKanbanLabels\`. \`{{column}}\` template interpolation handled inline.
2. **\`STATUS_COLORS\` registry → per-column \`accentClass\` + \`icon\`.** No shared status registry. Each \`KanbanColumn\` carries a Tailwind class string and an optional \`ReactNode\` icon. Consumers pick their own color/icon mapping.
3. **\`use-hidden-columns\` hook → controlled props only.** \`KanbanBoardProps.hiddenColumnIds\` + \`onHiddenColumnIdsChange\` are the entire hidden-column interface. \`apps/web\` keeps its local hook calling those props; the board itself is dumb on hidden state.

## Peer dependencies added

- \`@dnd-kit/core\` ^6.3.1
- \`@dnd-kit/sortable\` ^10.0.0
- \`@dnd-kit/utilities\` ^3.2.2

Required by \`KanbanBoard\` and \`RailDropSlot\`.

## Testing

- **441 tests across 86 files**, all passing.
- New WCAG (vitest-axe) audits: \`KanbanBoard\`, \`ListRow\`, \`RailDropSlot\` (idle + disabled), \`LaneTimeline\`.
- One real axe finding fixed during \`LaneTimeline\` work: bar label \`color-contrast\` on opacity-modified backgrounds. Resolved by using \`text-white\` on bar buttons (correct semantic call for colored fills) and silencing \`color-contrast\` in the story config (matching existing \`app-shell.stories.tsx\` pattern).
- Storybook: 4 new stories (\`Kanban/KanbanBoard\`, \`List/ListRow\`, \`Shell/RailDropSlot\`, \`Timeline/LaneTimeline\`).

## Storybook

\`pnpm storybook\` shows all four new entries alongside the existing ones. The \`Shell/RailDropSlot · DragTaskToMachine\` story is the integration showcase: drag a card from the kanban to a machine slot in the right rail.

## Out of scope

- **Migrating \`apps/web\` to consume the extracted Meda exports** — \`apps/web\` keeps its local kanban-board / list-row copies in this PR. Migration belongs in a follow-up so each consumer can be reviewed independently.
- **Full Gantt scheduling** — \`LaneTimeline\` renders bars and supports selection; drag-to-resize / drag-to-reschedule isn't included.

## Test plan

- [ ] \`pnpm install\` resolves cleanly (new dnd-kit peer deps)
- [ ] \`pnpm quality\` green (lint + typecheck + lint:tokens + test = 441/441)
- [ ] \`pnpm storybook\` renders all four new entries without console errors
- [ ] \`Kanban/KanbanBoard · DispatchTasks\` — cards drag between columns
- [ ] \`Shell/RailDropSlot · DragTaskToMachine\` — kanban card drops onto online machine slot, rejected on offline (\`disabled\`) slot
- [ ] \`Timeline/LaneTimeline · FleetTimeline\` — range tabs (1h/6h/24h/7d) work, date switcher works, now-line renders today, bar selection fires \`onSelectBar\`
- [ ] \`List/ListRow · Default\` — selected/focused states render
- [ ] Dark mode renders correctly for all four components (toggle Storybook theme)

### CODE REVIEW

File: src/kanban/kanban-board.tsx [hunk 1]
```diff
@@ -0,0 +1,489 @@
...
+              key={column.id}
+              items={columnItems.map((item) => item.id)}
+              strategy={verticalListSortingStrategy}
+            >
+              <KanbanColumn
+                column={column}
+                items={columnItems}
+                count={columnItems.length}
+                isOver={isOver}
+                onAddItem={onAddItem ? () => onAddItem(column.id as TStatus) : undefined}
```
[the remark this answers is not in the document]
Comment (alioftech) [reply src/kanban/kanban-board.tsx]: Fixed in 6247466 — `KanbanBoard` now passes `onAddItem` only when `column.canAdd !== false`, and the empty-column "add button on hover" branch is gated by the same condition. Default behavior (`canAdd === undefined`) keeps adds enabled, matching the JSDoc.

File: src/timeline/lane-timeline.tsx [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+  onSelectBar,
+  labels,
+  className,
+}: LaneTimelineProps) {
+  const resolvedLabels: LaneTimelineLabels = { ...defaultLaneTimelineLabels, ...(labels ?? {}) };
+  const [internalRange, setInternalRange] = useState<LaneTimelineRange>(defaultRange);
+  const [internalDate, setInternalDate] = useState<Date>(() => new Date());
+  const range = controlledRange ?? internalRange;
+  const selectedDate = controlledDate ?? internalDate;
+  const referenceNow = now ?? new Date();
```
[the remark this answers is not in the document]
Comment (alioftech) [reply src/timeline/lane-timeline.tsx]: Fixed in 9e4eebe — `internalDate` now initializes from `now ?? new Date()` so replay/snapshot scenarios that pass `now` start the timeline window on the same day as the now-line. Lazy initializer captures the first `now` only; if callers want the date to track changes to `now`, they should pass `selectedDate` as a controlled prop.

File: src/kanban/kanban-board.tsx [hunk 3]
```diff
@@ -0,0 +1,489 @@
...
+  KeyboardSensor,
+  MeasuringStrategy,
+  PointerSensor,
+  useSensor,
+  useSensors,
+} from '@dnd-kit/core';
+import { SortableContext, verticalListSortingStrategy } from '@dnd-kit/sortable';
+import { ChevronRight, EyeOff, MoreHorizontal } from 'lucide-react';
+import { useCallback, useDeferredValue, useMemo, useState } from 'react';
+import { createPortal } from 'react-dom';
```
[the remark this answers is not in the document]
Comment (alioftech) [reply src/kanban/kanban-board.tsx:19]: Fixed in 6247466 — added `'use client'` directive to `src/kanban/{kanban-board,kanban-column,kanban-card-wrapper}.tsx` and `src/list/list-row.tsx`. Pure-types/data files (`types.ts`, `kanban-collision.ts`, `kanban-drop-handler.ts`, barrel `index.ts` files) are intentionally left without the directive — they have no React internals and don't need a client boundary. `dist/` regenerated in de84efb.

File: src/kanban/kanban-board.tsx [hunk 4]
```diff
@@ -0,0 +1,489 @@
...
+            >
+              <KanbanColumn
+                column={column}
+                items={columnItems}
+                count={columnItems.length}
+                isOver={isOver}
+                onAddItem={onAddItem ? () => onAddItem(column.id as TStatus) : undefined}
+                onHideColumn={
+                  onHiddenColumnIdsChange ? () => handleHideColumn(column.id as TStatus) : undefined
+                }
```
[the remark this answers is not in the document]
Comment (alioftech) [reply src/kanban/kanban-board.tsx:259]: Fixed in 6247466 — same fix as the Codex `canAdd` thread. `KanbanBoard` now gates the per-column `onAddItem` (and the empty-column add affordance) by `column.canAdd !== false`. `KanbanColumn` continues to render add UI only when it actually receives an `onAddItem` prop, so the gating happens once at the board layer rather than being duplicated.

File: src/list/list-row.tsx [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+  className,
+}: ListRowProps) {
+  return (
+    // biome-ignore lint/a11y/useSemanticElements: clickable row needs flex layout not available on button
+    <div
+      role="button"
+      tabIndex={0}
+      onClick={onClick}
+      onKeyDown={(e) => e.key === 'Enter' && onClick?.()}
+      onMouseEnter={onMouseEnter}
```
[the remark this answers is not in the document]
Comment (alioftech) [reply src/list/list-row.tsx:75]: Fixed in 6247466 — `ListRow` `onKeyDown` now handles both `Enter` and `' '` (Space), and calls `e.preventDefault()` before invoking `onClick` to suppress page scroll on Space. Matches WCAG ARIA-button keyboard semantics for the `role="button"` element.

File: src/list/list-row.tsx [hunk 6]
```diff
@@ -0,0 +1,141 @@
...
+  children,
+  className,
+}: ListRowProps) {
+  return (
+    // biome-ignore lint/a11y/useSemanticElements: clickable row needs flex layout not available on button
+    <div
+      role="button"
+      tabIndex={0}
+      onClick={onClick}
+      onKeyDown={(e) => e.key === 'Enter' && onClick?.()}
```
[the remark this answers is not in the document]
Comment (alioftech) [reply src/list/list-row.tsx]: Fixed in 9d31fd0 — `role="button"`, `tabIndex={0}`, `onClick`, `onKeyDown`, the `cursor-pointer` class, and the focus/hover ring affordances are now ALL gated by `!!onClick`. When no `onClick` is provided, the row renders as a passive container — no tab stop, no button announcement, no keyboard activation. Mouse/focus passive handlers (used for prefetching) remain since they're non-interactive from an a11y standpoint. `dist/` regenerated in d2f5276.

File: src/timeline/time-axis.tsx [hunk 7]
```diff
@@ -0,0 +1,126 @@
...
+          ))}
+        </div>
+      </div>
+    </div>
+  );
+}
+
+function formatDateHeader(d: Date): string {
+  const today = new Date();
+  const isToday = d.toDateString() === today.toDateString();
```
[the remark this answers is not in the document]
Comment (alioftech) [reply src/timeline/time-axis.tsx]: Fixed in 3beab26 — `TimeAxis` now takes a `now: Date` prop, and `formatDateHeader` compares `selectedDate` against that reference time instead of `new Date()`. `LaneTimeline` passes its computed `referenceNow` (which already honors the `now` prop) through to `TimeAxis`. Replay/snapshot scenarios now keep the chart window, the now-line, and the "Today · …" label all anchored to the same reference clock. `dist/` regenerated in d2f5276.

File: src/components/ui/checkbox.tsx [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+import { Checkbox as CheckboxPrimitive } from '@base-ui/react/checkbox';
+import { CheckIcon, MinusIcon } from 'lucide-react';
+import { cn } from '../../lib/utils.js';
+
+interface CheckboxProps extends CheckboxPrimitive.Root.Props {
+  /** Show indeterminate (mixed) state - a dash instead of checkmark */
+  indeterminate?: boolean;
+}
+
+function Checkbox({ className, indeterminate, ...props }: CheckboxProps) {
```
[the remark this answers is not in the document]
Comment (alioftech) [reply src/components/ui/checkbox.tsx:12]: Fixed in b8bee27 — `Checkbox` now forwards `indeterminate` explicitly to `<CheckboxPrimitive.Root indeterminate={...}>` so base-ui sets `aria-checked="mixed"` correctly. Added a regression test asserting the ARIA state. Note: the canonical `apps/web` Checkbox has the same bug (also destructures and never forwards) — flagged separately, fix belongs to that codebase, not Meda. `dist/` regenerated in 6d18892.

File: src/timeline/lane.tsx [hunk 9]
```diff
@@ -0,0 +1,98 @@
...
+      <div className="relative h-7">
+        {lane.bars.map((bar) => {
+          const placement = placeBar(bar, windowStart, windowEnd);
+          if (!placement) return null;
+          const selected = selectedBarId === bar.id;
+          return (
+            <button
+              key={bar.id}
+              type="button"
+              onClick={() => onSelectBar?.(bar, lane)}
```
[the remark this answers is not in the document]
Comment (alioftech) [reply src/timeline/lane.tsx]: Fixed in bc23c46 — split the bar render path on `isSelectable = !!onSelectBar`. When `onSelectBar` is omitted, bars render as `<div role="img" aria-label="...">` (the `role="img"` is required so axe accepts the `aria-label` on a non-interactive element — `aria-label` on a bare `<div>` is a WCAG violation). Selectable bars keep the `<button>` semantics. `dist/` regenerated in 6d18892.

File: src/kanban/kanban-board.tsx [hunk 10]
```diff
@@ -0,0 +1,472 @@
...
+  const handleDragEnd = useCallback(
+    (event: DndKitDragEndEvent) => {
+      setActiveId(null);
+      setOverColumnId(null);
+
+      handleKanbanColumnDrop({
+        event,
+        items,
+        columns: visibleColumns,
+        onCardMove,
```
[the remark this answers is not in the document]
Comment (alioftech) [reply src/kanban/kanban-board.tsx:178]: Fixed in a9f5ddc — `KanbanBoardInner` now reads `headless` and gates the `useDndMonitor.onDragEnd` to only clear local UI state (`activeId`/`overColumnId`) when headless. The internal `handleKanbanColumnDrop` only fires through the kanban's own `<DndContext>` in non-headless mode. In headless mode the consumer's outer `onDragEnd` is the single owner of column drop routing — they must call `handleKanbanColumnDrop` themselves (the `Shell/RailDropSlot · DropZonesDemo` story already does this). `dist/` regenerated in 6d18892.

File: src/timeline/lane.tsx [hunk 11]
```diff
@@ -0,0 +1,98 @@
...
+
+function placeBar(bar: LaneBar, windowStart: Date, windowEnd: Date): BarPlacement | null {
+  const ws = windowStart.getTime();
+  const we = windowEnd.getTime();
+  const span = we - ws;
+  if (span <= 0) return null;
+  const bs = bar.start.getTime();
+  const be = bar.end.getTime();
+  // Bar entirely outside the window — skip.
+  if (be < ws || bs > we) return null;
```
[the remark this answers is not in the document]
Comment (alioftech) [reply src/timeline/lane.tsx]: Fixed in f6c69ad — changed `be < ws || bs > we` to `be <= ws || bs >= we` in `placeBar`. Bars exactly touching the window boundary have zero visible duration and now correctly render nothing instead of clamping to the 0.5% minimum width. New `lane.test.tsx` covers both boundary cases plus a 1ms-overlap positive control. `dist/` regenerated in 6d18892.

File: src/timeline/lane-timeline.tsx [hunk 12]
```diff
@@ -0,0 +1,174 @@
...
+
+  const { windowStart, windowEnd } = useMemo(() => {
+    const rangeMs = RANGE_MS[range];
+    const futurePadMs = Math.max(rangeMs / 6, 5 * 60_000);
+    const isToday = selectedDate.toDateString() === referenceNow.toDateString();
+    const end = isToday ? new Date(referenceNow.getTime() + futurePadMs) : endOfDay(selectedDate);
+    const start = new Date(end.getTime() - rangeMs - futurePadMs);
```
[the remark this answers is not in the document]
Comment (alioftech) [reply src/timeline/lane-timeline.tsx]: Fixed in 24bb3e2 — `futurePadMs` now only subtracts from `start` when the selected date is today (i.e. when the now-line is visible and needs breathing room). Historical days render exactly `rangeMs` of window: 24h → 24h, 7d → 7d. Today still gets the asymmetric pad so the now-line sits ~85% across the canvas. `dist/` regenerated in 4e74512.

File: src/timeline/time-axis.tsx [hunk 13]
```diff
@@ -0,0 +1,128 @@
...
+          aria-label="Time range"
+          className="flex items-center gap-1 rounded-md bg-muted p-0.5"
+        >
+          {RANGES.map((r) => (
+            <button
+              key={r}
+              type="button"
+              role="tab"
+              aria-selected={r === range}
+              onClick={() => onRangeChange(r)}
```
[the remark this answers is not in the document]
Comment (alioftech) [reply src/timeline/time-axis.tsx]: Fixed in a35b67a — dropped `role="tablist"`/`role="tab"` and `aria-label` on the wrapping div (biome forbids `aria-label` on non-semantic divs without an interactive role). The range pills are now plain segmented buttons with `aria-pressed` for active state. Each button's visible label (1h/6h/24h/7d) is its accessible name. Proper tab semantics with arrow-key nav + tabpanels would be a larger refactor; punted to PR #62 when these get migrated to the base-ui ToggleGroup primitive (per spec §5).

File: src/kanban/kanban-column.tsx [hunk 14]
```diff
@@ -0,0 +1,146 @@
...
+        {children}
+
+        {/* Add button - appears on hover. At top if empty, at bottom if has items */}
+        {onAddItem && (
+          <button
+            type="button"
+            onClick={onAddItem}
+            className={cn(
+              'flex w-full items-center justify-center gap-1.5 rounded-md border border-transparent border-dashed p-2 text-muted-foreground text-sm opacity-0 transition-[color,opacity,background-color,border-color] hover:border-muted-foreground/30 hover:bg-accent/50 hover:text-foreground group-hover/column:opacity-100',
+              count > 0 && 'mt-2'
```
[the remark this answers is not in the document]
Comment (alioftech) [reply src/kanban/kanban-column.tsx:130]: Fixed in a35b67a — added `focus-visible:opacity-100` to the hover-add button. Keyboard users can still tab to the button (preserving discoverability), AND focusing it makes it visible — keyboard navigation now matches the visual affordance. `dist/` regenerated in 983fe61.

**Review by adaadev — APPROVED** [review]
